simulations/vip-converter/vip-converter-testnet/simulations.ts

```diff
@@ -0,0 +1,363 @@
+      );
+
+      expect(await usdt.balanceOf(user1Address)).to.equal(usdtBalanceOfUserPrevious.sub(amountInMantissa));
+      expect(await RiskFundConverter.balanceOf(USDT)).to.equal(0);
+      expect(await RiskFundConverter.balanceOf(USDC)).to.equal(
+        usdcBalanceOfRiskFundConverterPrevious.sub(amountOutMantissa),
+      );
+      expect(await usdt.balanceOf(RISK_FUND_PROXY)).to.equal(
+        BigNumber.from(usdtBalanceOfRiskFundPrevious).add(amountInMantissa),
+      );
+    });
+
+    it("ConvertForExactTokens should work properly", async () => {
+      await usdc.connect(user1).allocateTo(user1Address, amount);
+      await usdc.connect(user1).transfer(RISK_FUND_CONVERTER, amount);
+      await RiskFundConverter.connect(user1).updateAssetsState(COMPTROLLER, USDC);
+
+      const amountOutMantissa = amount.div(2);
+      const amountInMaxMantissa = amount;
+      await usdt.connect(user1).allocateTo(user1Address, amount);
+
+      const usdcBalanceOfRiskFundConverterPrevious = await RiskFundConverter.balanceOf(USDC);
+      const usdtBalanceOfRiskFundPrevious = await usdt.balanceOf(RISK_FUND_PROXY);
+      const usdtBalanceOfUserPrevious = await usdt.balanceOf(user1Address);
+
+      const [, amountInMantissa] = await RiskFundConverter.connect(user1).getAmountIn(amountOutMantissa, USDT, USDC);
+
+      await usdt.connect(user1).approve(RISK_FUND_CONVERTER, amount);
+      await RiskFundConverter.connect(user1).convertForExactTokens(
+        amountInMaxMantissa,
+        amountOutMantissa,
+        USDT,
+        USDC,
+        user1Address,
+      );
+
+      expect(await usdt.balanceOf(user1Address)).to.equal(usdtBalanceOfUserPrevious.sub(amountInMantissa));
+      expect(await RiskFundConverter.balanceOf(USDT)).to.equal(0);
+      expect(await RiskFundConverter.balanceOf(USDC)).to.equal(
+        usdcBalanceOfRiskFundConverterPrevious.sub(amountOutMantissa),
+      );
+      expect(await usdt.balanceOf(RISK_FUND_PROXY)).to.equal(
+        BigNumber.from(usdtBalanceOfRiskFundPrevious).add(amountInMantissa),
+      );
+    });
+
+    it("Private conversion should occur on updateAssetsState", async () => {
+      const newAmount = amount.mul(2);
+      await usdc.connect(user1).allocateTo(user1Address, newAmount);
+      await usdt.connect(user1).allocateTo(user1Address, newAmount);
+
+      const destinationAddressForUsdcConverter = await USDCPrimeConverter.destinationAddress();
+
+      await usdc.connect(user1).transfer(RISK_FUND_CONVERTER, amount);
+      await RiskFundConverter.connect(user1).updateAssetsState(COMPTROLLER, USDC);
+
+      const plpBalanceForUsdcPrevious = await usdc.balanceOf(destinationAddressForUsdcConverter);
+
+      await usdt.connect(user1).transfer(USDC_PRIME_CONVERTER, amount);
+      // Private Conversion will occur
+      await USDCPrimeConverter.connect(user1).updateAssetsState(COMPTROLLER, USDT);
+
+      // New balance of plp will be updated by around 10 tokens because of minimum amount to convert.
```

Comment:
I don't think the issue is the $10 threshold, but the conversion rate USDC/USDT considering the oracle data

Reply:
Done https://github.com/VenusProtocol/vips/pull/49/commits/9eb73c07ce3c22d0e06dac9f5ba757469d673f1c